Mathlib/Analysis/Convex/Basic.lean

```diff
@@ -486,9 +491,32 @@ end AddCommGroup
 
 end OrderedRing
 
-section LinearOrderedSemiring
+section LinearOrder
+
+variable [Semiring 𝕜] [LinearOrder 𝕜] [AddCommMonoid E]
+
+section SemilinearMap
+
+variable {𝕜' : Type*} [Semiring 𝕜'] [LinearOrder 𝕜']
+variable {σ : 𝕜 →+* 𝕜'} [RingHomSurjective σ]
+variable {F' : Type*} [AddCommMonoid F'] [Module 𝕜' F'] [Module 𝕜 E]
+
+theorem Convex.semilinear_image {s : Set E} (hs : Convex 𝕜 s) (hσ : StrictMono σ)
+    (f : E →ₛₗ[σ] F') : Convex 𝕜' (f '' s) := by
+  rintro _ ⟨x, hx, rfl⟩ _ ⟨y, hy, rfl⟩ a b ha hb hab
+  obtain ⟨r, rfl⟩ : ∃ r : 𝕜, σ r = a := RingHomSurjective.is_surjective ..
+  obtain ⟨t, rfl⟩ : ∃ r : 𝕜, σ r = b := RingHomSurjective.is_surjective ..
+  refine ⟨r • x + t • y, hs hx hy (by simpa [← hσ.le_iff_le]) (by simpa [← hσ.le_iff_le])
+    ?_, by simp⟩
+  apply_fun σ using StrictMono.injective hσ
+  simpa
+
+theorem Convex.semilinear_range (hσ : StrictMono σ) (f : E →ₛₗ[σ] F') : Convex 𝕜' (Set.range f) :=
+  image_univ ▸ convex_univ.semilinear_image hσ f
```

Comment:
Same comment here: the approach via [Submodule.convex](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Analysis/Convex/Basic.html#Submodule.convex) needs no hypothesis on `σ`.